src/VirtoCommerce.ExperienceApiModule.DigitalCatalog/Schemas/ProductType.cs

```diff
@@ -27,24 +64,128 @@ public class ProductType : ObjectGraphType<ExpProduct>
-            Field(d => d.CatalogProduct.ImgSrc).Description("The product main image URL.");
+
+            Field<StringGraphType>(
+                "slug",
+                description: "Get slug for product.",
+                resolve: context =>
+                {
+                    var semanticUrl = context.Source.CatalogProduct?.SeoInfos?.FirstOrDefault()?.SemanticUrl;
+                    return semanticUrl;
+                });
+
+            Field<StringGraphType>(
+                "metaDescription",
+                description: "Get metaDescription for product.",
+                resolve: context =>
+                {
+                    var metaDescription = context.Source.CatalogProduct?.SeoInfos?.FirstOrDefault()?.MetaDescription;
+                    return metaDescription;
+                });
+
+            Field<StringGraphType>(
+                "metaKeywords",
+                description: "Get metaKeywords for product.",
+                resolve: context =>
+                {
+                    var metaKeywords = context.Source.CatalogProduct?.SeoInfos?.FirstOrDefault()?.MetaKeywords;
+                    return metaKeywords;
+                });
+
+            Field<StringGraphType>(
+                "metaTitle",
+                description: "Get metaTitle for product.",
+                resolve: context =>
+                {
+                    var metaTitle = context.Source.CatalogProduct?.SeoInfos?.FirstOrDefault()?.PageTitle;
+                    return metaTitle;
+                });
+
+            Field<StringGraphType>(
+                "imgSrc",
+                description: "The product main image URL.",
+                resolve: context => context.Source.CatalogProduct.ImgSrc);
+
             Field(d => d.CatalogProduct.OuterId, nullable: true).Description("The outer identifier");
-            Field<ListGraphType<PropertyType>>("properties", resolve: context => context.Source.CatalogProduct.Properties);
 
-            Field<ListGraphType<PriceType>>("prices", arguments: new QueryArguments(new QueryArgument<StringGraphType> { Name = "currency", Description = "currency" }), resolve: context =>
+            Field<StringGraphType>(
+                "brandName",
+                description: "Get brandName for product.",
+                resolve: context =>
+                {
+                    var brandName = context.Source.CatalogProduct.Properties
+                        ?.FirstOrDefault(x => x.Name == "Brand")
+                        ?.Values
+                        ?.FirstOrDefault(x => x.Value != null)
+                        ?.Value;
+
+                    return brandName;
+                });
+
+            Field<VariationType>("masterVariation", resolve: context =>
             {
-                var result = (context.Source).Prices;
-                var currency = context.GetArgument<string>("currency");
-                if (currency != null)
+                var variation = new Variation();
```

Comment:
If `Product` has set `MainProductId` property it can be considered as `masterVariation` otherwise need to resolve  main product with a separate call